sources/SilkTouch/SilkTouch/JobContext.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Silk.NET.SilkTouch
+{
+    /// <summary>
+    /// Information on the current Job
+    /// </summary>
+    public class JobContext
```

Comment:
Let's make this internal for now given that it's only our CLI that uses it today and there's no immediate use case for anyone else using this.

Reply:
Can't, makes DI upset for the progressService. You would have to make the constructor internal as well and DI doesn't seem to like that. It only seems to work with a public one.

Reply:
Still think this should be possible somehow, but happy to leave as is for now. Use this (from #2427):
```cs
[Experimental(
    "ST0005",
    UrlFormat = "https://dotnet.github.io/Silk.NET/docs/v3/silk.net/diagnostics/{0}"
)]
```

Reply:
Oh wait this is a different issue, just use the overload of AddSingleton that takes a delegate instead of relying on the activator to instantiate the class.

Reply:
Set now to internal since no public references are made to it